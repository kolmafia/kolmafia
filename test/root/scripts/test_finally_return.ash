int test( int input )
{
    try {
        print("You gave me " + input);
    } finally {
        if (input == 100) {
            int output = input + 1;
            print("I prefer " + output);
            return output;
        }
	if (input == 200) {
	    abort("No way!");
	}
    }
    print("Good enough!");
    return input;
}

print("I received " + test(50));
print("I received " + test(100));
print("I received " + test(200));