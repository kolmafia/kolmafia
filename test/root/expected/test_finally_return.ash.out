You gave me 50
Good enough!
I received 50
You gave me 100
I prefer 101
I received 101
You gave me 200
No way!
